git/git.go

```diff
@@ -27,12 +27,16 @@ func Dir() (string, error) {
 	if cachedDir != "" {
 		return cachedDir, nil
 	}
-
-	dirCmd := gitCmd("rev-parse", "-q", "--git-dir")
-	dirCmd.Stderr = nil
-	output, err := dirCmd.Output()
+	gitDir, err := CommondirName()
 	if err != nil {
-		return "", fmt.Errorf("Not a git repository (or any of the parent directories): .git")
+		// Fall back for older version of git prior to v2.7.0 (added worktrees)
+		dirCmd := gitCmd("rev-parse", "--git-dir")
+		dirCmd.Stderr = nil
+		output, err := dirCmd.Output()
+		if err != nil {
+			return "", fmt.Errorf("Not a git repository (or any of the parent directories): .git")
+		}
+		gitDir = firstLine(output)
```

Comment:
@mislav Thanks for the responses!  I added this function. Thoughts being this perhaps isn't the last time a ```hub``` command has different behavior in a worktree.

https://github.com/github/hub/blob/479ed29d9253c88f4da78caf8c9061a6da760528/git/git.go#L82

In ```Dir()```, if the function returns an error, we fall back to the original behavior of the code